### PR TITLE
boards: native: guard motor driver simulation code

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

native has some simulated "motor driver" support, which is *always* compiled in. This PR guards it on ~~`MODULE_MOTOR_DRIVER`~~`MODULE_PERIPH_QDEC`.

### CODE REVIEW

File: boards/native/include/board.h [hunk 1]
```diff
@@ -127,6 +127,7 @@ extern mtd_dev_t *mtd0;
 /** @} */
 #endif
 
+#ifdef MODULE_MOTOR_DRIVER
```
Comment (fjmolinas) [inline boards/native/include/board.h:130]: Would have preferred an `IS_USED` but not worth re-triggering the ci for this.

**Review by fjmolinas — APPROVED** [review]
ACK

File: boards/native/include/board.h (hunk 1 above)
Comment (fjmolinas) [reply boards/native/include/board.h:130]: Ah well build failed, so this could be addressed

Comment (kaspar030) [reply boards/native/include/board.h:130]: yup.

**Comment by kaspar030** [comment]
```
In file included from /home/runner/work/RIOT/RIOT/boards/native/board_init.c:17:
/home/runner/work/RIOT/RIOT/boards/native/include/board.h:130:12: error: missing binary operator before token "("
  130 | #if IS_USED(MODULE_PERIPH_QDEC)
      |            ^
make[2]: *** [/home/runner/work/RIOT/RIOT/Makefile.base:142: /home/runner/work/RIOT/RIOT/examples/hello-world/bin/native/board/board_init.o] Error 1
make[1]: *** [/home/runner/work/RIOT/RIOT/Makefile.base:31: ALL--/home/runner/work/RIOT/RIOT/boards/native] Error 2
make: *** [/home/runner/work/RIOT/RIOT/examples/hello-world/../../Makefile.include:686: application_hello-world.module] Error ```

huh? that's from the "tools-test" ci run?

**Comment by aabadie** [comment]
Murdock is reporting the same error. Apparently in riotctrl, the hello-world application is built for native.

**Pull request merged**